Tests/Segment-Tests/Analytics_Tests.swift

```diff
@@ -123,6 +123,12 @@ final class Analytics_Tests: XCTestCase {
         let outputReader = OutputReaderPlugin()
         analytics.add(plugin: outputReader)
         
+        // prime the pump for userAgent, since it's retrieved async.
+        let vendorSystem = VendorSystem.current
+        while vendorSystem.userAgent == nil {
+            RunLoop.main.run(until: Date.distantPast)
```

Comment:
Is this a tight loop? basically spinning until that variable is set? If so, does a something like a count down latch fit more here?